presto-main/src/main/java/com/facebook/presto/operator/ByteArrayUtils.java

```diff
@@ -0,0 +1,205 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import io.airlift.slice.Slice;
+import io.airlift.slice.SliceOutput;
+import sun.misc.Unsafe;
+
+import java.lang.reflect.Field;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static sun.misc.Unsafe.ARRAY_BYTE_BASE_OFFSET;
+import static sun.misc.Unsafe.ARRAY_INT_INDEX_SCALE;
+import static sun.misc.Unsafe.ARRAY_LONG_INDEX_SCALE;
+
+public class ByteArrayUtils
+{
+    static Unsafe unsafe;
+
+    static {
+        try {
+            Field field = Unsafe.class.getDeclaredField("theUnsafe");
+            field.setAccessible(true);
+            unsafe = (Unsafe) field.get(null);
+            if (unsafe == null) {
+                throw new RuntimeException("Unsafe access not available");
+            }
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private ByteArrayUtils() {}
+
+    public static int writeInts(byte[] target, int targetIndex, int[] values, int offset, int length)
+    {
+        try {
+            checkArgument(targetIndex + length * ARRAY_INT_INDEX_SCALE <= target.length || targetIndex >= 0);
+            checkArgument(offset + length <= values.length);
+        }
+        catch (Exception e) {
+            e.printStackTrace();
```

Comment:
This will print to stderr and pollute the logs

Reply:
Ah sorry forgot to remove it